packages/backend/src/registries/LocalRepositoryRegistry.ts

```diff
@@ -36,16 +42,35 @@ export class LocalRepositoryRegistry extends Publisher<LocalRepository[]> {
     this.notify();
 
     return Disposable.create(() => {
-      this.unregister(localRepository.path);
+      this.unregister(localRepository.path).catch((reason: unknown) => console.log(String(reason)));
     });
   }
 
-  unregister(path: string): void {
+  async unregister(path: string): Promise<void> {
+    await this.deleteLocalRepository(path);
     this.repositories.delete(path);
     this.notify();
   }
 
+  async deleteLocalRepository(path: string): Promise<void> {
+    return fs.promises.rm(path, { recursive: true, force: true, maxRetries: 3 });
+  }
+
   getLocalRepositories(): LocalRepository[] {
     return Array.from(this.repositories.values());
   }
+
+  loadLocalRecipeRepositories(recipes: Recipe[]): void {
```

Comment:
Could we have this inside an `init` method ? so it matches other behaviour. I guess this is used to solve https://github.com/containers/podman-desktop-extension-ai-lab/issues/358 ?